### ISSUE TITLE
symbol(s) not found for architecture x86_64

### ISSUE THREAD

**Issue by ghost** [nirai-compiler/src#14]
Hi. Trying to compile my game on OS X Yosemite 10.10, and I'm getting this error with pythonembed.

```
Thomass-MacBook-Pro:build thomas$ ppython -OO make.py -c
Compiling CXX codes...
Compiling... src/rebuilt.cxx
Compiling... /Users/thomas/ToontownRebuiltSource/build/nirai/src/main.cxx
Compiling... /Users/thomas/ToontownRebuiltSource/build/nirai/src/aes.cxx
Linking...
Undefined symbols for architecture x86_64:
  "_init_subprocess", referenced from:
      __PyImport_Inittab in libpythonembed.a(config.o)
  "_initnt", referenced from:
      __PyImport_Inittab in libpythonembed.a(config.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
The following command returned non-zero value (1): g++ -o built/rebuilt -L"/Users/thomas/ToontownRebuiltSource/build/nirai/python" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/vorbis/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/freetype/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/squish/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/tiff/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/png/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/openssl/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/built/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/jpeg/lib" -L"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/thirdparty/darwin-libs-a/ode/lib" -F"/Users/thomas/ToontownRebuiltSource/build/nirai/panda3d/built/Frameworks" -framework AGL -framework OpenGL -framework Cg -framework Cocoa -framework OpenAL -framework Carbon -framework AppKit -lp3direct -lpandatiffxx -ljpeg -lpanda -lp3framework -lp3tinydisplay -lvorbisenc -lp3dtoolconfig -lsquish -l_panda3d_fx -lpng -lvorbis -lpythonembed -lvorbisfile -lpandaphysics -lode -logg -lp3interrogatedb -l_panda3d_core -lpandagl -lpandafx -l_panda3d_physics -l_panda3d_interrogatedb -lpandaode -lcrypto -lp3dtool -l_panda3d_ode -l_panda3d_direct -lssl -lpandaexpress -lpandaegg -l_panda3d_egg -lpandatiff -lz -lfreetype -lp3openal_audio "libpandadna/DNASuitPoint.o" "libpandadna/DNAFlatBuilding.o" "built/rebuilt.o" "libpandadna/DNAProp.o" "libpandadna/DNALandmarkBuilding.o" "libpandadna/DNACornice.o" "libpandadna/libpandadna_module.o" "libpandadna/suit_igate.o" "libpandadna/DNASign.o" "libpandadna/DNADoor.o" "libpandadna/DNAVisGroup.o" "libpandadna/components_igate.o" "libpandadna/SuitLegList.o" "libpandadna/DNASuitPath.o" "libpandadna/config_dna.o" "built/main.o" "libpandadna/DNABattleCell.o" "built/aes.o" "libpandadna/DNANode.o" "libpandadna/DNASignBaseline.o" "libpandadna/SuitLeg.o" "libpandadna/DNASuitEdge.o" "libpandadna/DNASignGraphic.o" "libpandadna/DNAStorage.o" "libpandadna/DNAStreet.o" "libpandadna/DNAWall.o" "libpandadna/DNAAnimProp.o" "libpandadna/DNAGroup.o" "libpandadna/base_igate.o" "libpandadna/DNAFlatDoor.o" "libpandadna/DNAWindows.o" "libpandadna/DNALoader.o" "libpandadna/DNAInteractiveProp.o" "libpandadna/DNAAnimBuilding.o" -dylib_file /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
```

How do I fix this?

**Comment by DarthMDev** [comment]
i get different one
Undefined symbols for architecture x86_64:
  "init_libmotiontrail()", referenced from:
      _main in main.o
  "_initdirect", referenced from:
      _main in main.o
  "_initode", referenced from:
      _main in main.o
  "_initphysics", referenced from:
      _main in main.o
ld: symbol(s) not found for architecture x86_64

**Comment by mfwass** [comment]
Recompile libpythonembed, I'm thinking it was compiled on another machine or a previous OS.  You may need to replace the OpenSSL libraries to be compatible with your OS.

**Comment by ghost** [comment]
I'm pretty sure I compiled the Nirai python on my mac, but this was a year ago so I wouldn't remember. I've been searching for a couple days for OpenSSL libraries for Mac.

I tried using brew, then `brew link openssl` but it told me to pass some includes to the compiler (When panda3d's compile.sh failed, it did say the command that didn't return 0 was clang++ or something), but I can't find where I'd add the options it said (something like -I/usr/opt/openssl, but there were two options) to the command. (Sorry if that was hard to follow.)

Could you please point me in the right direction (if you can)?

**Comment by mfwass** [comment]
Take the libs from the newer SSL and place them in the `thirdparty` SSL location.

**Comment by ghost** [comment]
So, I had to reinstall OS X (this time, I'm on 10.11 El Capitan) because I somehow screwed something up. I rebuilt Nirai following the guide for Mac in the readme file. Now, I have the same error, but it appears to be a different cause. 

```
Thomass-MacBook-Pro:build thomas$ ppython -OO make.py -c
:interrogatedb(warning): Class panda3d.core.PythonThread has a zero TypeHandle value; check that init_type() is called.
:interrogatedb(warning): Class panda3d.core.PythonTask has a zero TypeHandle value; check that init_type() is called.
:interrogatedb(warning): Class panda3d.core.ShaderBuffer has a zero TypeHandle value; check that init_type() is called.
:interrogatedb(error): Attempt to use type none which has not yet been defined!
Compiling CXX codes...
Compiling... src/rebuilt.cxx
In file included from src/rebuilt.cxx:2:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/datagram.h:20:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:35:46: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  MutexPosixImpl(const MutexPosixImpl &copy) DELETED;
                                             ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from src/rebuilt.cxx:2:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/datagram.h:20:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:36:59: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  MutexPosixImpl &operator = (const MutexPosixImpl &copy) DELETED_ASSIGN;
                                                          ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:259:26: note: 
      expanded from macro 'DELETED_ASSIGN'
#  define DELETED_ASSIGN DELETED
                         ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from src/rebuilt.cxx:2:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/datagram.h:20:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:63:50: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  ReMutexPosixImpl(const ReMutexPosixImpl &copy) DELETED;
                                                 ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from src/rebuilt.cxx:2:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/datagram.h:20:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:64:63: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  ReMutexPosixImpl &operator = (const ReMutexPosixImpl &copy) DELETED;
                                                              ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from src/rebuilt.cxx:2:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/datagram.h:23:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pta_uchar.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerToArray.h:62:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerToArrayBase.h:20:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerTo.h:61:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerToBase.h:18:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerToVoid.h:39:44: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  PointerToVoid(const PointerToVoid &copy) DELETED;
                                           ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
5 warnings generated.
Compiling... /Users/thomas/Documents/client-source-mac/build/nirai/src/aes.cxx
Compiling... /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx:3:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipeSelection.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipe.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsDevice.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedReferenceCount.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:35:46: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  MutexPosixImpl(const MutexPosixImpl &copy) DELETED;
                                             ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx:3:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipeSelection.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipe.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsDevice.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedReferenceCount.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:36:59: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  MutexPosixImpl &operator = (const MutexPosixImpl &copy) DELETED_ASSIGN;
                                                          ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:259:26: note: 
      expanded from macro 'DELETED_ASSIGN'
#  define DELETED_ASSIGN DELETED
                         ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx:3:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipeSelection.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipe.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsDevice.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedReferenceCount.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:63:50: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  ReMutexPosixImpl(const ReMutexPosixImpl &copy) DELETED;
                                                 ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx:3:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipeSelection.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipe.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsDevice.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedReferenceCount.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typedObject.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeHandle.h:166:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/typeRegistry.h:18:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexImpl.h:42:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/mutexPosixImpl.h:64:63: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  ReMutexPosixImpl &operator = (const ReMutexPosixImpl &copy) DELETED;
                                                              ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx:3:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipeSelection.h:19:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipe.h:21:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerTo.h:61:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerToBase.h:18:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/pointerToVoid.h:39:44: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  PointerToVoid(const PointerToVoid &copy) DELETED;
                                           ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx:3:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipeSelection.h:19:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipe.h:56:42: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  GraphicsPipe(const GraphicsPipe &copy) DELETED;
                                         ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/src/main.cxx:3:
In file included from /Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipeSelection.h:19:
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/graphicsPipe.h:57:55: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  GraphicsPipe &operator = (const GraphicsPipe &copy) DELETED_ASSIGN;
                                                      ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:259:26: note: 
      expanded from macro 'DELETED_ASSIGN'
#  define DELETED_ASSIGN DELETED
                         ^
/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/include/dtoolbase_cc.h:177:24: note: 
      expanded from macro 'DELETED'
#     define DELETED = delete
                       ^
7 warnings generated.
Linking...
Undefined symbols for architecture x86_64:
  "_EVP_MD_CTX_free", referenced from:
      Multifile::Subfile::write_data(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, std::__1::basic_istream<char, std::__1::char_traits<char> >*, std::__1::fpos<__mbstate_t>, Multifile*) in libp3openal_audio.a(p3express_composite1.o)
      ConfigPage::read_prc(std::__1::basic_istream<char, std::__1::char_traits<char> >&) in libp3openal_audio.a(p3prc_composite1.o)
  "_EVP_MD_CTX_new", referenced from:
      Multifile::check_signatures() in libp3openal_audio.a(p3express_composite1.o)
      Multifile::Subfile::write_data(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, std::__1::basic_istream<char, std::__1::char_traits<char> >*, std::__1::fpos<__mbstate_t>, Multifile*) in libp3openal_audio.a(p3express_composite1.o)
      ConfigPage::read_prc(std::__1::basic_istream<char, std::__1::char_traits<char> >&) in libp3openal_audio.a(p3prc_composite1.o)
  "_OPENSSL_init_crypto", referenced from:
      OpenSSLWrapper::OpenSSLWrapper() in libp3openal_audio.a(p3express_composite1.o)
      EncryptStreamBuf::open_read(std::__1::basic_istream<char, std::__1::char_traits<char> >*, bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in libp3openal_audio.a(p3prc_composite2.o)
      EncryptStreamBuf::open_write(std::__1::basic_ostream<char, std::__1::char_traits<char> >*, bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in libp3openal_audio.a(p3prc_composite2.o)
  "_OPENSSL_init_ssl", referenced from:
      OpenSSLWrapper::notify_ssl_errors() in libp3openal_audio.a(p3express_composite1.o)
      OpenSSLWrapper::OpenSSLWrapper() in libp3openal_audio.a(p3express_composite1.o)
      OpenSSLWrapper::notify_debug_ssl_errors() in libp3openal_audio.a(p3express_composite1.o)
  "_OPENSSL_sk_free", referenced from:
      Multifile::validate_signature_certificate(int) const in libp3openal_audio.a(p3express_composite1.o)
  "_OPENSSL_sk_new", referenced from:
      Multifile::validate_signature_certificate(int) const in libp3openal_audio.a(p3express_composite1.o)
  "_OPENSSL_sk_num", referenced from:
      OpenSSLWrapper::load_certificates_from_pem_ram(char const*, unsigned long) in libp3openal_audio.a(p3express_composite1.o)
      HTTPChannel::validate_server_name(x509_st*) in libp3openal_audio.a(p3downloader_composite2.o)
  "_OPENSSL_sk_pop_free", referenced from:
      OpenSSLWrapper::load_certificates_from_pem_ram(char const*, unsigned long) in libp3openal_audio.a(p3express_composite1.o)
  "_OPENSSL_sk_push", referenced from:
      Multifile::validate_signature_certificate(int) const in libp3openal_audio.a(p3express_composite1.o)
  "_OPENSSL_sk_value", referenced from:
      OpenSSLWrapper::load_certificates_from_pem_ram(char const*, unsigned long) in libp3openal_audio.a(p3express_composite1.o)
      HTTPChannel::validate_server_name(x509_st*) in libp3openal_audio.a(p3downloader_composite2.o)
  "_TLS_client_method", referenced from:
      HTTPClient::get_ssl_ctx() in libp3openal_audio.a(p3downloader_composite2.o)
  "_X509_STORE_up_ref", referenced from:
      HTTPClient::get_ssl_ctx() in libp3openal_audio.a(p3downloader_composite2.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
The following command returned non-zero value (1): g++ -o built/rebuilt -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/squish/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/python" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/jpeg/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/tiff/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/ode/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/png/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/openssl/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/freetype/lib" -L"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/thirdparty/darwin-libs-a/vorbis/lib" -F"/Users/thomas/Documents/client-source-mac/build/nirai/panda3d/built/Frameworks" -framework AGL -framework OpenGL -framework Cg -framework Cocoa -framework OpenAL -framework Carbon -framework AppKit -ljpeg -lssl -lp3openal_audio -logg -l_panda3d_core -lpng -lpandagl -lp3dtoolconfig -l_panda3d_physics -lvorbis -l_panda3d_ode -lpandaode -lpandaphysics -lpandatiff -lp3dtool -lvorbisfile -lcrypto -l_panda3d_direct -lfreetype -lpandaexpress -lpandaegg -lpythonembed -lp3tinydisplay -lp3framework -lpanda -lp3direct -lp3interrogatedb -lpandatiffxx -l_panda3d_egg -l_panda3d_interrogatedb -lode -lsquish -lz -l_panda3d_fx -lpandafx -lvorbisenc "built/main.o" "built/rebuilt.o" "built/aes.o" -dylib_file /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
```

From this log, can you spot anything I might have done wrong?

**Comment by ghost** [comment]
I rebuilt panda3d and for some reason, this time it worked. I've now compiled the game fully, there's just one more issue I need to work out. It's a seperate issue, so I'll close this one. I think it has something to do with Nirai because of reasons I'll explain in the new issue.
